File: app/src/main/java/com/nikola/jakshic/truesight/view/HeroSortDialog.java
package com.nikola.jakshic.truesight.view;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.nikola.jakshic.truesight.R;

public class HeroSortDialog extends DialogFragment {

    public HeroSortDialog() {
    }

    public interface OnSortListener {
        void onSort(int sortOption);
    }

    public static HeroSortDialog newInstance() {
        HeroSortDialog dialog = new HeroSortDialog();
        return dialog;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setTitle("Sort by")
                .setItems(R.array.sort_hero_options, (dialog, which) ->
                    ((OnSortListener) getTargetFragment()).onSort(which))
                .create();
    }
}

File: app/src/main/java/com/nikola/jakshic/dagger/ui/profile/heroes/HeroSortDialog.kt
package com.nikola.jakshic.dagger.ui.profile.heroes

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.nikola.jakshic.dagger.R

class HeroSortDialog : DialogFragment() {

    interface OnSortListener {
        fun onSort(sort: Int)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(context!!)
                .setTitle(getString(R.string.sort_by))
                .setItems(R.array.sort_hero_options) { _, which ->
                    (targetFragment as OnSortListener).onSort(which)
                }.create()
    }
}